tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,566 @@
+#    distribution is indistinguishable from zero) is obtained by
+#    permutations. This means that it makes no assumptions of Gaussianity
+#    (which do hold for this example but do not in general).
+
+titles.append('$\mathbf{Perm_{max}}$')
+out = permutation_t_test(X)[:2]
+ts.append(out[0])
+ps.append(out[1])
+mccs.append(True)
+plot_t_p(ts[-1], ps[-1], titles[-1], mccs[-1])
+
+###############################################################################
+# Clustering
+# ^^^^^^^^^^
+# Each of the aforementioned multiple comparisons corrections have the
+# disadvantage of not fully incorporating the correlation structure of the
+# data. However, by defining the connectivity/adjacency/neighbor structure
+# in our data, we can use **clustering** to compensate.
+#
+# To use this, we need to rethink our null hypothesis. Instead
+# of thinking about a null hypothesis about means per voxel, we consider a
+# null hypothesis about sizes of clusters in our data, which could be stated
+# like:
+#
+#     The distribution of spatial cluster sizes observed in two experimental
+#     conditions are drawn from the same probability distribution.
+#
+# Here we only have a single condition and we contrast to zero, which can
+# be thought of as:
+#
+#     The distribution of spatial cluster sizes is independent of the sign
+#     of the data.
+#
+# In this case, we again do permutations with a maximum statistic, but, under
+# each permutation, we:
+#
+# 1. Threshold the computed statistic with some **initial**
+#    ``threshold`` value.
+# 2. Cluster points that exceed this threshold (with the same sign)
+#    based on adjacency.
+# 3. Record the *size* of each cluster (measured, e.g., by a simple vertex
+#    count or the sum of voxel t-values within the cluster).
+#
+# After doing these permutations, the cluster sizes in our veridical data
+# are compared to this null distribution. The p-value associated with each
+# cluster is again given by the proportion of smaller null distribution
+# values. This can then be subjected to a standard p-value threshold
+# (e.g., ``p < 0.05``) to reject the null hypothesis (i.e., find an effect
+# of interest).
+#
+# This reframing to consider *cluster sizes* rather than *individual means*
+# maintains the advantages of the standard non-parametric permutation
+# test -- namely controlling FWER and making no assumptions of parametric
+# data distribution.
+# Cricitally, though, it also accounts for the correlation structure in the
+# data -- which in this toy case is spatial but
+# in general can be multidimensional (e.g., spatio-temporal) -- because the
+# null distribution will be derived from data in a way that preserves these
+# correlations.
+#
+# However, there is a drawback. If a cluster significantly deviates from
+# the null, no further inference on the cluster (e.g., peak location) can be
+# made, as the entire cluster as a whole is used to reject the null.
```

Comment:
phrased differently, because the test statistic concerns the full data, and the null hypothesis thus also refers to the (cluster structure of the) full data

Reply:
nice! this also helps clarify the relationship to http://www.fieldtriptoolbox.org/faq/how_not_to_interpret_results_from_a_cluster-based_permutation_test (which I'll also link to)